src/components/PostDetails.tsx

```diff
@@ -1,117 +1,132 @@
+  const [sending, setSending] = useState<boolean>(false);
+
+  useEffect(() => {
+    setComments(() => null);
+    setOpenCommentForm(() => false);
+
+    getComments(id)
+      .then((response) => {
+        setComments(() => response);
+        setCommentsListerror(false);
+      })
+      .catch(() => {
+        setCommentsListerror(true);
+        setComments(() => []);
+      });
+  }, [id]);
+
+  const handleRemoveComment = (commentId: number) => {
+    setComments((prev) => prev?.filter(
+      (item) => item.id !== commentId,
+    ) ?? null);
+
+    removeComment(commentId);
+  };
+
+  const handleAddComment = (newComment: CommentData) => {
+    setSending(true);
+
+    addComment({ ...newComment, postId: id })
+      .then((response) => {
+        setComments((prev) => (prev !== null ? [...prev, response]
+          : [response]));
+      }).finally(() => setSending(false));
+  };
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            {`#${id}: ${title}`}
           </h2>
 
           <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
+            {body}
           </p>
         </div>
 
         <div className="block">
-          <Loader />
+          {!commentsListerror && !comments && (<Loader />)}
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+          {commentsListerror && (
+            <div className="notification is-danger" data-cy="CommentsError">
+              Something went wrong
+            </div>
+          )}
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+          {!commentsListerror && comments && comments.length === 0 && (
+            <p className="title is-4" data-cy="NoCommentsMessage">
+              No comments yet
+            </p>
+          )}
 
-          <p className="title is-4">Comments:</p>
-
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+          {comments && comments.length > 0 && (
+            <>
+              <p className="title is-4">Comments:</p>
+              {comments.map((comment) => (
```

Comment:
Use a destruction for `comment`

Reply:
destructor can not be use here couse id attribute is already on top of this file